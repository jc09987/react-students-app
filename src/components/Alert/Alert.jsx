import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

/**
 * Alert to display system messages
 */
class Alert extends PureComponent {
  render = () => {
    const {
      id, className, title, text
    } = this.props;
    return (
      <div id={id} className={className} role="alert">
        <strong>{title}</strong>
        {' '}
        {text}
.
      </div>
    );
  }
}

Alert.defaultProps = {
  text: ''
};

Alert.propTypes = {
  /** Alert id */
  id: PropTypes.string.isRequired,

  /** Alert CSS class */
  className: PropTypes.string.isRequired,

  /** Alert title */
  title: PropTypes.string.isRequired,

  /** Alert text */
  text: PropTypes.string
};

export default Alert;
