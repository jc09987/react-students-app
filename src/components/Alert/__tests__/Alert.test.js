import renderer from 'react-test-renderer';
import React from 'react';
import Alert from '../Alert';

describe('Alert component renders correctly (info)', () => {
    it('renders correctly', () => { 
        const rendered = renderer.create(
            <Alert
                id="info-alert"
                className="alert alert-info"
                title="Loading..."
                text="Fetching data..."
            />
        );
        expect(rendered.toJSON()).toMatchSnapshot();
    });
});

describe('Alert component renders correctly (error)', () => {
    it('renders correctly', () => { 
        const rendered = renderer.create(
            <Alert
                id="error-alert"
                className="alert alert-danger"
                title="Error!"
                text="oh snap!"
            />
        );
        expect(rendered.toJSON()).toMatchSnapshot();
    });
});