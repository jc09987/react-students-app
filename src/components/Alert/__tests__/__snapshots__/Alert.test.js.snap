// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component renders correctly (error) renders correctly 1`] = `
<div
  className="alert alert-danger"
  id="error-alert"
  role="alert"
>
  <strong>
    Error!
  </strong>
   
  oh snap!
  .
</div>
`;

exports[`Alert component renders correctly (info) renders correctly 1`] = `
<div
  className="alert alert-info"
  id="info-alert"
  role="alert"
>
  <strong>
    Loading...
  </strong>
   
  Fetching data...
  .
</div>
`;
